src/fsharp/CompileOps.fs

```diff
@@ -1639,15 +1639,111 @@ let GetDefaultFSharpCoreReference() = typeof<list<int>>.Assembly.Location
 
 // If necessary assume a reference to the latest System.ValueTuple with which those tools are built.
 let GetDefaultSystemValueTupleReference() = 
-    try 
-       let asm = typeof<System.ValueTuple<int,int>>.Assembly
-       if asm.FullName.StartsWith "System.ValueTuple" then 
-           Some asm.Location 
-       else None
-    with _ -> None
+    let path = Path.GetDirectoryName(GetDefaultFSharpCoreReference())
```

Comment:
This is not really right and the existing code is wrong too ... 

This forces us to co-locate System.ValueTuple.dll with FSharp.Core.dll  which is not right.  Especially since dotnet cli hosting can locate assemblies in a central shared cache rather than app local.

The existing code assumes that ValueTuple was loaded from an assembly named System.ValueTuple.dll which is also not right.  Since from dotnet 4.7 it exists in mscorlib, probably System.Private.Corlib.dll on coreclr.

I guess I will have to revisit this code at some point, but we need to do better long term.